assets/js/components/Link.js

```diff
@@ -35,13 +40,29 @@ function Link( {
 	caps,
 	danger,
 	disabled,
+	ariaLabel,
 	...extraProps
 } ) {
 	// Note: the disabled attribute does not alter behavior of anchor tags,
 	// so if disabled we force it to be a button.
 	const isAnchor = href && ! disabled;
 	const SemanticLink = isAnchor ? 'a' : 'button';
 
+	const getAriaLabel = () => {
+		let label = ariaLabel;
+		const newTabLabel = _x( '(opens in a new tab)', 'screen reader text', 'google-site-kit' );
+
+		if ( ariaLabel === '' && external && typeof children === 'string' ) {
+			label = children;
+		}
+
+		if ( external ) {
+			label += ` ${ newTabLabel }`;
+		}
+
+		return label;
```

Comment:
This could be simplified by checking `external` first and if not, return `ariaLabel` right away since there's no benefit to duplicating the string text content as the `aria-label` otherwise. Then the two current `external` checks could be removed.

Reply:
@aaemnnosttv We should cater for the scenario where both, ariaLabel is provided and the link is external. So, even if the ariaLabel is supplied, if it's external, we should append the `(opens in a new tab) ` text to the passed on ariaLabel prop. Whereas what you're suggesting is that is that we just check whether we check for external, if not, return ariaLabel.

Reply:
@asvinb I think you misunderstood my comment. I agree we should support a provided `ariaLabel` if the link is `external` as well. My point was that the function could return early if `external === false` and then the remaining conditionals (for when `external` is true) could be simplified, and the second conditional would no longer be needed and the `newTabLabel` would be unconditionally appended.

Something like this:
```js
let label = ariaLabel;

if ( ! external ) {
	return label;
} else if ( typeof children === 'string' ) {
	label = label || children;
}

return `${ label } ${ _x( '(opens in a new tab)', 'screen reader text', 'google-site-kit' ) }`;
```

Reply:
Ah now I see what you meant. Updated @aaemnnosttv